nrcemt/qeels/test/test_peak_detection.py

```diff
@@ -160,19 +164,153 @@ def test_mark_peaks():
+#     result = rotate_spectrogram(spectrogram, 0.4896955931291964)
 
-    # # 0.4896955931291964
-    # np.testing.assert_array_almost_equal(result, rotated)
+#     # 0.4896955931291964
+#     np.testing.assert_array_almost_equal(result, rotated)
 
-
-#     plasmon_array = [
+#     plasmon_array=[
 #         [913, 217], [917, 685],
 #         [0, 0], [0, 0],
 #         [0, 0], [0, 0]
-#     ]
+#
+
+
+def test_calculation_e():
+    dirname = os.path.dirname(__file__)
+    peak_x_path = os.path.join(dirname, 'resources/peak_pos_x.mat')
+    peak_x = loadmat(peak_x_path)
+    peak_x = peak_x['Peak_position_x'].flatten()
+
+    res = calculation_e(200, peak_x)
+    assert res == 4139449
+
+
+def test_bulk_calculations():
+    dirname = os.path.dirname(__file__)
+    peak_x_path = os.path.join(dirname, 'resources/peak_pos_x.mat')
+    peak_y_path = os.path.join(dirname, 'resources/peak_pos_y.mat')
+    image2_path = os.path.join(dirname, 'resources/image2.mat')
+
+    image2 = loadmat(image2_path)
+    image2 = image2['image2']
+
+    spectrogram_path = os.path.join(dirname, 'resources/Converted.prz')
+    spectrogram = load_spectrogram(spectrogram_path)
+
+    peak_x = loadmat(peak_x_path)
+    peak_x = peak_x['Peak_position_x'].flatten()
+
+    peak_y = loadmat(peak_y_path)
+    peak_y = peak_y['Peak_position_y'].flatten()
+
+    result, image = bulk_calculations(peak_x, peak_y, 15.0, spectrogram)
+
+    np.testing.assert_almost_equal(result, 0.051094149613447)
+    np.testing.assert_allclose(
+        image, image2,
+        atol=10000, rtol=1
+    )
+
+
+# 1:(682, 482), 2:(844, 390)
+def test_calculation_q():
+    dirname = os.path.dirname(__file__)
+    peak_x_path = os.path.join(dirname, 'resources/peak_pos_x(upper).mat')
+    peak_y_path = os.path.join(dirname, 'resources/peak_pos_y(upper).mat')
+
+    peak_x = loadmat(peak_x_path)
+    peak_x = peak_x['Peak_position_x'].flatten()
+
+    peak_y = loadmat(peak_y_path)
+    peak_y = peak_y['Peak_position_y'].flatten()
+
+    res = calculation_q(
+        1165934, peak_x, peak_y, 0.0569
+    )
+
+    res2 = calculation_q(
+        3.127567785110500e5,
+        peak_x, peak_y,
+        0.0569
+    )
+
+    res3 = calculation_q(
+        1.5243e5,
+        peak_x, peak_y,
+        0.0569
+    )
+
+    res4 = calculation_q(
+        1.6345e5,
+        peak_x, peak_y,
+        0.0569
+    )
+
+    res5 = calculation_q(
+        1.9546e5,
+        peak_x, peak_y,
+        0.0569
+    )
+
+    res6 = calculation_q(
+        2e6,
+        peak_x, peak_y,
+        0.0569
+    )
+
+    np.testing.assert_almost_equal(res, 1.613344424025299e03)
+    np.testing.assert_almost_equal(res2, 1.483953382190898e03)
+    np.testing.assert_almost_equal(res3, 1.872686170354184e+03)
+    np.testing.assert_almost_equal(res4, 1.787347483147202e+03)
+    np.testing.assert_almost_equal(res5, 1.624980374906233e+03)
+    np.testing.assert_almost_equal(res6, 1.628523280306290e+03)
```

Comment:
instead of doing stuff like this I think it makes a more sense to keep the result closer to the actual function call it matches. eg.
```
np.testing.assert_almost_equal(
    calculation_q(
         3.127567785110500e5,
         peak_x, peak_y,
         0.0569
    ),
    1.613344424025299e03
)